02-peer-protocol.md

```diff
@@ -1183,42 +1189,53 @@ sign the resulting transaction (as defined in [BOLT #3](03-transactions.md)), an
    * [`u16`:`num_htlcs`]
    * [`num_htlcs*signature`:`htlc_signature`]
 
+1. `tlv_stream`: `commitment_signed_tlvs`
+2. types:
+    1. type: 1 (`alternative_feerates`)
+    2. data:
+        * [`...*u32`:`feerates_per_kw`]
+
 #### Requirements
 
 A sending node:
   - MUST NOT send a `commitment_signed` message that does not include any
-updates.
-  - MAY send a `commitment_signed` message that only
-alters the fee.
-  - MAY send a `commitment_signed` message that doesn't
-change the commitment transaction aside from the new revocation number
-(due to dust, identical HTLC replacement, or insignificant or multiple
-fee changes).
+    updates.
+  - MAY send a `commitment_signed` message that only alters the fee.
+  - MAY send a `commitment_signed` message that doesn't change the
+    commitment transaction aside from the new revocation number (due to dust,
+    identical HTLC replacement, or insignificant or multiple fee changes).
   - MUST include one `htlc_signature` for every HTLC transaction corresponding
     to the ordering of the commitment transaction (see [BOLT #3](03-transactions.md#transaction-input-and-output-ordering)).
+  - If `option_anchors` and `option_alternative_feerates` were negotiated:
+    - MUST send `alternative_feerates` containing other feerates for which it
+      will send signatures.
+    - MUST send one `alternative_commitment_signed` for each of those feerates.
```

Comment:
I find it weird that Alice decides what feerates will be available to Bob in Bob's commitment tx, and vice-versa. Ideally it would be Alice who can decide for herself that she wants feerates `[3,10,30,100,300]` (sat/vbyte). What if Alice sends some nice rates to Bob but Bob does not give any alternatives to Alice (`alternative_feerates=[]`)? Is Alice supposed to force-close because of not having enough feerate options?

Reply:
I agree, that's one of the decisions I wanted to have on this PR, thanks for raising it :wink:. What we do here depends on whether we do "broadcaster pays" or not and whether we re-sign commitment transactions or not.

One issue with Alice deciding what feerates Bob should sign for her commit transaction is that she ask Bob to pay a feerate so high that HTLCs all go to dust, and then she could broadcast the corresponding commitment once it's revoked and Bob won't be able to claim anything. Similarly to what we have with `update_fee`, Bob shouldn't sign feerates that it feels are absurdly high. But we can probably find a good middle ground that works well enough in practice.